### PR TITLE
6 get/api/users

### PR DESCRIPTION
**Pull request by MrMcSnail**
Adds a new /api/users endpoint. 
A get request responds with an array of all user objects from the DB.

### CODE REVIEW

File: MODELS/comments.models.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+const db = require('../db/connection');
```
Comment (cjplatten) [inline MODELS/comments.models.js:1]: Be careful when making commits that you only add the necessary files, this file is not needed for this commit!

File: __tests__/app.test.js [hunk 2]
```diff
@@ -105,3 +105,20 @@ describe('PATCH /api/articles/:article_id', () => {
...
 });
+
+describe('GET /api/users', ()=>{
+  test(`Status 200: Responds with an array of objects, each object should have the following property:
+username`, () => {
+    return request(app).get('/api/users').expect(200).then(({body})=>{
+      expect(body.users).toBeInstanceOf(Array);
+      body.users.forEach((user)=>{
+        expect(user).toHaveProperty('username');
+      });
```
Comment (cjplatten) [inline __tests__/app.test.js:116]: what response would you get from this test if the body.users array was empty?

File: __tests__/app.test.js [hunk 3]
```diff
@@ -105,3 +105,20 @@ describe('PATCH /api/articles/:article_id', () => {
...
+    });
+  });
+  test(`Status 404: Mis-spelled url`, () => {
+    return request(app).get('/api/handbag').expect(404).then(({body})=>{
+      expect(body).toHaveProperty('message', 'Not Found');
+    });
```
Comment (cjplatten) [inline __tests__/app.test.js:122]: Is this test specific to this endpoint?

**Review by cjplatten — CHANGES_REQUESTED** [review]
A couple of comments to look at before merging, otherwise loking good!

File: __tests__/app.test.js (hunk 3 above)
Comment (MrMcSnail) [reply __tests__/app.test.js:122]: Probably not neccessary at all in fact. My 404 error in my app will trigger the same response

**Pull request merged**